pkg/virtctl/network/dynamicifaces_test.go

```diff
@@ -0,0 +1,146 @@
+)
+
+var _ = Describe("Dynamic Interface Attachment", func() {
+	var (
+		ctrl         *gomock.Controller
+		kubeClient   *fakek8sclient.Clientset
+		vmInterface  *kubecli.MockVirtualMachineInterface
+		vmiInterface *kubecli.MockVirtualMachineInstanceInterface
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
+		kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+
+		kubeClient = fakek8sclient.NewSimpleClientset()
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	const (
+		ifaceName   = "pluggediface1"
+		networkName = "newnet"
+		vmName      = "myvm1"
+	)
+
+	mockVMIAddInterfaceEndpoints := func(vmName string, networkName string, ifaceName string) {
+		kubecli.MockKubevirtClientInstance.
+			EXPECT().
+			VirtualMachineInstance(k8smetav1.NamespaceDefault).
+			Return(vmiInterface).
+			Times(1)
+		vmiInterface.EXPECT().AddInterface(vmName, gomock.Any()).DoAndReturn(func(arg0, arg1 interface{}) interface{} {
+			Expect(arg1.(*v1.AddInterfaceOptions).NetworkName).To(Equal(networkName))
+			Expect(arg1.(*v1.AddInterfaceOptions).InterfaceName).To(Equal(ifaceName))
+			return nil
+		})
+	}
+
+	mockVMAddInterfaceEndpoints := func(vmName string, networkName string, ifaceName string) {
+		kubecli.MockKubevirtClientInstance.
+			EXPECT().
+			VirtualMachine(k8smetav1.NamespaceDefault).
+			Return(vmInterface).
+			Times(1)
+		vmInterface.EXPECT().AddInterface(vmName, gomock.Any()).DoAndReturn(func(arg0, arg1 interface{}) interface{} {
+			Expect(arg1.(*v1.AddInterfaceOptions).NetworkName).To(Equal(networkName))
+			Expect(arg1.(*v1.AddInterfaceOptions).InterfaceName).To(Equal(ifaceName))
+			return nil
+		})
+	}
+
+	DescribeTable("should fail when required input parameters are missing", func(cmdType string, args ...string) {
+		cmd := clientcmd.NewVirtctlCommand(append([]string{cmdType}, args...)...)
+		Expect(cmd.Execute()).To(HaveOccurred())
+	},
+		Entry("missing the VM name as parameter for the `AddInterface` cmd", network.HotplugCmdName),
+		Entry("missing all required flags for the `AddInterface` cmd", network.HotplugCmdName, vmName),
+		Entry("missing the network name flag for the `AddInterface` cmd", network.HotplugCmdName, vmName, "--iface-name", ifaceName),
+		Entry("missing the interface name flag for the `AddInterface` cmd", network.HotplugCmdName, vmName, "--network-name", networkName),
+	)
+
+	When("all the required input parameters are provided", func() {
+		var requiredCmdArgs []string
+
+		BeforeEach(func() {
+			kubeClient.Fake.PrependReactor("*", "*", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+				Expect(action).To(BeNil())
+				return true, nil, errors.New("kubeClient command not mocked")
+			})
+
+			requiredCmdArgs = []string{"--network-name", networkName, "--iface-name", ifaceName}
+		})
+
+		isPersistent := func(flags []string) bool {
```

Comment:
IMO there are too many usages of closures in the tests, making it hard to read them cleanly.
It also has the potential to drag with it bugs (using a closure instead of a passed in variable and vice versa.